models/ncnewsmodels.js

```diff
@@ -21,22 +21,39 @@ exports.selectArticleById = (article_id) => {
 
   exports.selectAllArticles = () => {
     const selectArticlesQuery = format(
-      `SELECT
+      `SELECT 
       articles.article_id,
       articles.title,
       articles.topic,
       articles.author,
       articles.created_at,
       articles.votes,
+      articles.article_img_url,
       COUNT(comments.comment_id) AS comment_count
-    FROM articles
-    LEFT JOIN comments ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC;
+      FROM articles
+      JOIN comments ON articles.article_id = comments.article_id
+      GROUP BY articles.article_id
+      ORDER BY created_at DESC;
     `,
     );
-    return db.query(selectArticlesQuery).then((result) => {
+    return db.query(selectArticlesQuery).then((result) => { 
     return(result.rows)
    });
   };
 
+  exports.selectAllComments = (id) => {
+    const selectCommentsQuery = format(
+      `SELECT * FROM comments WHERE article_id = %L
+      ORDER BY created_at DESC;`, id
+    );
+    return db.query(selectCommentsQuery).then(({rows}) => { 
+      if (rows.length === 0) {
+      return rows
+    }
```

Comment:
You have two tests that are getting to this point and returning rows if it's empty, the /api/articles/999/comments and /api/articles/2/comments. 

We have different responses we want to give to these tests. 

999 isn't an article that exists, so we want to return an error with a promise.reject saying the article doesn't exist. 

2 is an article that exists, so in that case we want to return the empty rows. 

You will have to query the database here and check that the article exists. If it does, return the emtpy array. If it doesn't, return the error.